contracts/Core/AssetManager.sol

```diff
@@ -64,22 +64,35 @@ contract AssetManager is ACL, AssetStorage {
 
     event CollectionActivityStatus(uint256 id, uint256 epoch, bool active, uint256 timestamp);
 
-    constructor(address parametersAddress) {
+    modifier checkState(uint256 state) {
+        require(state == parameters.getState(), "incorrect state");
+        _;
+    }
+
+    modifier checkStates(uint256 state1, uint256 state2) {
+        require(state1 == parameters.getState() || state2 == parameters.getState(), "incorrect states");
+        _;
+    }
+
+    constructor(address parametersAddress, address blockManagerAddress) {
         parameters = IParameters(parametersAddress);
+        blockManager = IBlockManager(blockManagerAddress);
     }
 
     function createJob(
         string calldata url,
         string calldata selector,
-        string calldata name,
-        bool repeat
+        string calldata name
     ) external onlyRole(parameters.getAssetModifierHash()) {
-        numAssets = numAssets + 1;
         uint256 epoch = parameters.getEpoch();
-        Structs.Job memory job = Structs.Job(numAssets, epoch, url, selector, name, repeat, true, msg.sender, 0, uint256(assetTypes.Job));
-        jobs[numAssets] = job;
+        require(numAssetsCreated[epoch] < parameters.maxAssetsCreatedPerEpoch(), "Asset Creation Limit Reached");
```

Comment:
Hey why do we have this limit

Reply:
This update has been made on the basis of this discussion
https://github.com/razor-network/contracts/pull/241#discussion_r677310860

Reply:
If its required we dont have to store numAssetsCreated every epoch IMO
we can define it as global variable 
and reset it to 0 in confirm

Reply:
Noted

Reply:
Okay the issue i thought of doing this way was if no one confirms the block, then the reset wont take place until someone commits

Reply:
Hmm yes, maybe a solution we can find
First, let's see if it's needed, I am not sure if we should carry this overhead
Let's discuss it in the next call